File: java-engine/src/main/java/engine/algorithms/faulty_dfs_dispersion/models/FaultyDfsDispersionEdge.java
package engine.algorithms.faulty_dfs_dispersion.models;

import engine.core.models.Edge;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class FaultyDfsDispersionEdge extends Edge {

    private int fromPort;
    private int toPort;
    
}

File: dispersion-app/src/app/models/algorithms/faulty-dfs-dispersion/FaultyDfsDispersionEdge.ts
import { Edge } from '../../core/Edge';

export class FaultyDfsDispersionEdge extends Edge {
  fromPort: number;
  toPort: number;

  constructor(
    id: number,
    fromID: number,
    toID: number,
    fromPort: number,
    toPort: number
  ) {
    super(id, fromID, toID);
    this.fromPort = fromPort;
    this.toPort = toPort;
  }
}
